juju/juju
# add generation data for each model to .local

## Description of change

Keep track of the model's generation per user in .local.  The generation commands will update it appropriately.  The ModelGeneration() can be used by all commands using ModelCommandBase to get the current-model generation. 

## QA steps

1. export JUJU_DEV_FEATURE_FLAGS=generations
2. juju bootstrap localhost
3. check ~/.local/share/juju/models.yaml, generation for each model should be 'current'.
3. juju add-model one
3.  check ~/.local/share/juju/models.yaml, generation for model 'one' should be 'current'.
5. juju add-generation
3.  check ~/.local/share/juju/models.yaml, generation for model 'one' should be 'next'.
4. juju switch-generation current
3.  check ~/.local/share/juju/models.yaml, generation for model 'one' should be 'current'.
4. juju switch-generation next
3.  check ~/.local/share/juju/models.yaml, generation for model 'one' should be 'next'.
5. juju cancel-generation
3.  check ~/.local/share/juju/models.yaml, generation for model 'one' should be 'current'.
3. juju deploy percona-cluster mysql
3. juju add-generation
3.  check ~/.local/share/juju/models.yaml, generation for model 'one' should be 'next'.
3. juju config mysql  <- should end with note that changes will be to next generation.

## Documentation changes

In the future.
